plugin/storage/es/factory.go

```diff
@@ -190,32 +189,36 @@ func createSpanWriter(
 }
 
 // GetSpanServiceMappings returns span and service mappings
-func GetSpanServiceMappings(shards, replicas int64, esVersion uint) (string, string) {
+func GetSpanServiceMappings(shards, replicas int64, esVersion uint, esPrefix string) (string, string) {
 	if esVersion == 7 {
-		return fixMapping(loadMapping("/jaeger-span-7.json"), shards, replicas),
-			fixMapping(loadMapping("/jaeger-service-7.json"), shards, replicas)
+		return fixMapping(loadMapping("/jaeger-span-7.json"), shards, replicas, esPrefix),
+			fixMapping(loadMapping("/jaeger-service-7.json"), shards, replicas, esPrefix)
 	}
-	return fixMapping(loadMapping("/jaeger-span.json"), shards, replicas),
-		fixMapping(loadMapping("/jaeger-service.json"), shards, replicas)
+	return fixMapping(loadMapping("/jaeger-span.json"), shards, replicas, ""),
+		fixMapping(loadMapping("/jaeger-service.json"), shards, replicas, "")
 }
 
 // GetDependenciesMappings returns dependencies mappings
 func GetDependenciesMappings(shards, replicas int64, esVersion uint) string {
 	if esVersion == 7 {
-		return fixMapping(loadMapping("/jaeger-dependencies-7.json"), shards, replicas)
+		return fixMapping(loadMapping("/jaeger-dependencies-7.json"), shards, replicas, "")
 	}
-	return fixMapping(loadMapping("/jaeger-dependencies.json"), shards, replicas)
+	return fixMapping(loadMapping("/jaeger-dependencies.json"), shards, replicas, "")
 }
 
 func loadMapping(name string) string {
 	s, _ := mappings.FSString(false, name)
 	return s
 }
 
-func fixMapping(mapping string, shards, replicas int64) string {
-	mapping = strings.Replace(mapping, "${__NUMBER_OF_SHARDS__}", strconv.FormatInt(shards, 10), 1)
-	mapping = strings.Replace(mapping, "${__NUMBER_OF_REPLICAS__}", strconv.FormatInt(replicas, 10), 1)
-	return mapping
+func fixMapping(mapping string, shards, replicas int64, esPrefix string) string {
+	t, _ := pongo2.FromString(mapping)
+	if esPrefix != "" {
+		esPrefix += "-"
+	}
+	fixedMapping, _ := t.Execute(pongo2.Context{"NumberOfShards": shards, "NumberOfReplicas": replicas, "ESPrefix": esPrefix, "UseILM": false, "Order": 1})
```

Comment:
@albertteoh  - I did add the flag (useILM) to collector in initial commits and then reverted it. My thought process behind it below:

- If we do have a flag to enable UseILM in jaeger collector - we have to add a check if its being used along with --use-aliases or not. Because without aliases - there is no use of enabling ILM and it might need lead to confusion. 
- If you think about it, the actual support for ILM (creation of initial indices, aliases ) is done by rollover.py. It makes no difference if have the flag at collector or not.

Reply:
> we have to add a check if its being used along with --use-aliases or not

I thought this should be a fairly easy and cheap check, though maybe I'm wrong?

> If you think about it, the actual support for ILM (creation of initial indices, aliases ) is done by rollover.py. It makes no difference if have the flag at collector or not.

Consider these two scenarios where I think supporting `use-ilm` in jaeger-collector would be useful:
- On-boarding to ILM from manual rollovers (`init` with ILM disabled, but start jaeger-collector with ILM enabled), which simply involves a single manual rollover to pickup the ILM-enabled template.
- Off-boarding from ILM to manual rollovers (see https://github.com/jaegertracing/jaeger/pull/2454#discussion_r539078122).

Further, I think it would reduce confusion for developers, who see that `UseILM = false` in jaeger-collector, yet wonder why ILM is working; as what happened to me when reviewing this PR :).

Reply:
Yep - I see your point 👍 . Let me revert my commits which actually added support for this flag along with tests and push. Would work on this during weekend.

Reply:
Added UseILM flag to es storage config and updated corresponding tests. Kindly check @albertteoh